master.go

```diff
@@ -919,7 +920,193 @@ func newMasterCollector(httpClient *httpClient) prometheus.Collector {
+			re, _ := regexp.Compile("master/frameworks/(.*?)/tasks/task_(.*?)")
+			for metric, value := range m {
+				matches := re.FindStringSubmatch(metric)
+				if len(matches) != 3 {
+					continue
+				}
+				id := matches[1]
+				state := matches[2]
+
+				if !strings.HasSuffix(state, "ing") {
+					continue
+				}
+				c.(*settableCounterVec).Set(value, id, state)
+			}
+			return nil
+		},
+
+		// Framework task terminal states
+		counter("master", "frameworks_task_states_terminal", "Terminal task states per framework", "framework_id", "state"): func(m metricMap, c prometheus.Collector) error {
+			re, _ := regexp.Compile("master/frameworks/(.*?)/tasks/task_(.*?)")
+			for metric, value := range m {
+				matches := re.FindStringSubmatch(metric)
+				if len(matches) != 3 {
+					continue
+				}
+				id := matches[1]
+				state := matches[2]
+
+				if strings.HasSuffix(state, "ing") {
+					continue
+				}
+				c.(*settableCounterVec).Set(value, id, state)
+			}
+			return nil
+		},
+
+		// Framework offers
+		counter("master", "frameworks_offers", "Number of offers by type per framework", "framework_id", "type"): func(m metricMap, c prometheus.Collector) error {
+			re, _ := regexp.Compile("master/frameworks/(.*?)/offers/(.*?)")
+			for metric, value := range m {
+				matches := re.FindStringSubmatch(metric)
+				if len(matches) != 3 {
+					continue
+				}
+				id := matches[1]
+				typ := matches[2]
+				c.(*settableCounterVec).Set(value, id, typ)
+			}
+			return nil
+		},
+
+		// Per-framework resource allocation: offer filters
+		gauge("master", "frameworks_offer_filters", "Number of filters received per framework over period", "framework_id", "period"): func(m metricMap, c prometheus.Collector) error {
+			re, _ := regexp.Compile("master/frameworks/(.*?)/offer_filters/refuse_seconds/(.*?)")
+			for metric, value := range m {
+				matches := re.FindStringSubmatch(metric)
+				if len(matches) != 3 {
+					continue
+				}
+				id := matches[1]
+				pd := matches[2]
+				c.(*settableCounterVec).Set(value, id, pd)
+			}
+			return nil
+		},
+
+		gauge("master", "frameworks_resource_filters", "Number of times resources were filtered per reason", "framework_id", "reason"): func(m metricMap, c prometheus.Collector) error {
+			re, _ := regexp.Compile("master/frameworks/(.*?)/resources_filtered/(.*?)")
+			for metric, value := range m {
+				matches := re.FindStringSubmatch(metric)
+				if len(matches) != 3 {
+					continue
+				}
+				id := matches[1]
+				why := matches[2]
+				c.(*settableCounterVec).Set(value, id, why)
+			}
+			return nil
+		},
+
+		// Per-role resource allocation (min)
+		gauge("master", "frameworks_role_resource_allocation_min", "Number of times resources were filtered per role", "framework_id", "role"): func(m metricMap, c prometheus.Collector) error {
+			re, _ := regexp.Compile("master/frameworks/(.*?)/roles/(.*)/latest_position/min")
```

Comment:
The right format is:

`master/frameworks/<framework id>.<framework name>/allocation/roles/<role name/latest_position/min`